tests/D2L.CodeStyle.Analyzers.Test/Specs/TypeDeclarationImmutabilityAnalyzer.cs

```diff
@@ -0,0 +1,598 @@
+﻿// analyzer: D2L.CodeStyle.Analyzers.Immutability.TypeDeclarationImmutabilityAnalyzer

```

Comment:
Taking a little extra time to look at these; writing some notes by hand about what I think needs to be tested and comparing.

I find these tests good but typically "write/append-only", so this is the only way I can think to review it.

Reply:
Yeah. Some of the groupings are a little inconsistent, largely due to impatience and evolution over time, but the idea was to just try to exercise all the patterns for each "type", instead of assuming/knowing that read-only field checks are done top-level and only need to be tested once, for instance.

It's annoying, but trying to blackbox it.

Was theoretically going to copy-paste them to be in a `struct` and an `[ImmutableBaseClass]` as well 😧

Reply:
Need to add an auto-implemented `set` for properties.